File: src/main/java/com/github/kayjamlang/core/expressions/ValueExpression.java
package com.github.kayjamlang.core.expressions;

import com.github.kayjamlang.core.opcodes.AccessType;

public class ValueExpression extends Expression {

    public final Object value;

    public ValueExpression(Object value) {
        super(AccessType.NONE, -1);
        this.value = value;
    }

    @Override
    public String toString() {
        return value.toString();
    }
}

File: src/main/scala/com/github/kayjamlang/core/expressions/ValueExpression.scala
package com.github.kayjamlang.core.expressions

import com.github.kayjamlang.core.Type
import com.github.kayjamlang.core.opcodes.AccessType

class ValueExpression(val value: Any, val valueType: Type) extends Expression(AccessType NONE, -1) {
  override def toString: String = value toString
}
